iib/workers/tasks/build.py

```diff
@@ -720,6 +725,69 @@ def _verify_index_image(
         )
 
 
+def _add_property_to_index(db_path, property):
+    """
+    Add a property to the index.
+
+    :param str db_path: path to the index database
+    :param dict property: a dict representing a property to be added to the index.db
+    """
+    insert = (
+        'INSERT INTO properties '
+        '(type, value, operatorbundle_name, operatorbundle_version, operatorbundle_path) '
+        'VALUES (?, ?, ?, ?, ?);'
+    )
+    con = sqlite3.connect(db_path)
+    # Insert property
+    con.execute(
+        insert,
+        (
+            property['type'],
+            property['value'],
+            property['operatorbundle_name'],
+            property['operatorbundle_version'],
+            property['operatorbundle_path'],
+        ),
+    )
+    con.commit()
+    con.close()
+
+
+def _requires_max_ocp_version(bundle):
+    """
+    Check if the bundle requires the olm.maxOpenShiftVersion property.
+
+    This property is required for bundles using deprecated APIs that don't already
+    have the olm.maxOpenShiftVersion property set.
+
+    :param str bundle: a string representing the bundle pull specification
+    :returns bool:
+    """
+    cmd = [
+        'operator-sdk',
+        'bundle',
+        'validate',
+        bundle,
+        '--select-optional',
+        'name=community',
+        '--output=json-alpha1',
+        '--image-builder',
+        'none',
+    ]
+    result = run_cmd(cmd, strict=False)
+    if result:
+        output = json.loads(result)
+        # check if the bundle validation failed
+        if not output['passed']:
+            # check if the failure is due to the presence of deprecated APIs
+            # and absence of the 'olm.maxOpenShiftVersion' property
+            # Note: there is no other error in the sdk that mentions this field
+            for msg in output['outputs']:
+                if 'olm.maxOpenShiftVersion' in msg['message']:
+                    return True
+    return False
```

Comment:
HI @gallettilance, 

Be aware that the message will be different for each case scenario, Ditto, so it is missing to work properly with. See that in the following cases we ought to not overwritten the info provided by the author:

1.- If the bundle has deprecated APIs and the property is informed and with a value informed which is >= 4.9 
```
 {
            "type": "error",
            "message": "Error: Value : (etcdoperator.v0.9.4) csv.Annotations.olm.properties with the key and value for olm.maxOpenShiftVersion has the OCP version value 4.9 which is \u003e= of 4.9. This bundle is using APIs which were deprecated and removed in v1.22. More info: https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-22. Migrate the API(s) for CRD: ([\"etcdbackups.etcd.database.coreos.com\" \"etcdclusters.etcd.database.coreos.com\" \"etcdrestores.etcd.database.coreos.com\"]) or inform in this property an OCP version which is \u003c 4.9"
        }
```

2.- If the bundle has deprecated APIs and the property is informed and has an invalid value (attention to this scenario)```
  {
            "type": "error",
            "message": "Error: Value : (etcdoperator.v0.9.4) csv.Annotations.olm.properties has an invalid value.Unable to parse (invalid) using semver : Invalid character(s) found in major number \"invalid\""
        }
```

Reply:
discussed offline: we will track these scenarios as bugs / enhancements for the coming sprint. cc @yashvardhannanavati